sdv/model_generator/tree_generator/file_import.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from typing import List
+
+from sdv.model_generator.tree_generator.constants import JSON, VSPEC
+from sdv.model_generator.tree_generator.file_formats import Json, Vspec, formats
+
+
+# if no other file supported format is found
+class UnsupportedFileFormat(Exception):
+    def __init__(self, *args, **kwargs):
+        self.file_name = args[0]
+        self.line_nr = args[1]
+        self.message = args[2]
+        Exception.__init__(self, *args, **kwargs)
+
+    def __str__(self):
+        return "{}: {}: {}".format(self.file_name, self.line_nr, self.message)
+
+
+class FileImport:
+    def __init__(self, file_path: str, include_dirs: List, strict, overlays):
+        self.file_path = file_path
+        self.include_dirs = include_dirs
+        self.strict = strict
+        self.overlays = overlays
+        # setting the format from the file_path
+        self.file_format = self.__get_format(self.file_path)
+
+    def __get_format(self, file_path: str):
+        for format in formats:
+            if re.match(r"^.+\." + f"{format}" + r"$", file_path):
```

Comment:
No heavy regex needed, Python's `os` lib can do this out of the box:
`file_ext = os.path.splitext('path/to/my/file.txt') # file_ext will be '.txt'`

Moreover you can reduce boilerplate code here (for loop is not needed):

```
file_ext = os.path.splitext(file_path)[1:]
if file_ext in formats:
  if file_ext == VSPEC:
    ...
  elif file_ext == JSON:
    ...
else:
  raise UnsupportedFileFormat(...)
```